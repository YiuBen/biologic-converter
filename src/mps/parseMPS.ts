import { TextData } from 'cheminfo-types';
import { ensureString } from 'ensure-string';

export function parseMPS(arrayBuffer: TextData) {
  const lines = ensureString(arrayBuffer, {
    encoding: 'latin1',
  }).split(/\r?\n/);
}
