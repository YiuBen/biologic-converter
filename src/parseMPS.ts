export async function parseMPS(file: File) {
  const content = await file.text();
}
