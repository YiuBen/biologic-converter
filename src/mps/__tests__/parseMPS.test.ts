import { readFileSync } from 'fs';
import { join } from 'path';

import { parseMPS } from '../parseMPS';

describe('parseMPS', () => {
  it('test', () => {
    const arrayBuffer = readFileSync(join(__dirname, 'data/test/test.mps'));
    parseMPS(arrayBuffer);
  });
});
