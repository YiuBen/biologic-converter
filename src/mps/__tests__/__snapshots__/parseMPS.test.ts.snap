// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseMPS test 1`] = `
Object {
  "Cable": "standard",
  "Channel": "Floating",
  "Characteristic mass": "0.001 g",
  "Comments": "",
  "Cycle Definition": "Charge/Discharge alternance
Turn to OCV between techniques",
  "Density": "0.000 g/cm3",
  "Device": "SP-200",
  "Electrode connection": "standard",
  "Electrode material": "",
  "Electrode surface area": "0.001 cmÂ²",
  "Electrolyte": "",
  "Equivalent Weight": "0.000 g/eq.",
  "Ewe ctrl range": "min = -10.00 V, max = 10.00 V",
  "Ewe,I filtering": "50 kHz",
  "Filename": "Z:\\\\Data group member\\\\Anna\\\\EC\\\\2021-10-20_AL0006_SolGel4_EC1_-0.5V_ref1021_1\\\\CA_AL0006_.mps",
  "Initial state": "",
  "Number of linked techniques": "2",
  "Technique": Object {
    "1": Object {
      "Ru": "68.300",
      "comp. level (%)": "85",
      "comp. mode": "Software",
      "name": "Manual IR compensation",
      "unit Ru": "Ohm",
    },
    "2": Object {
      "Bandwidth": "8",
      "E range max (V)": "10.000",
      "E range min (V)": "-10.000",
      "Ei (V)": "-0.400",
      "I Range": "10 mA",
      "I Range init": "Unset",
      "I Range max": "Unset",
      "I Range min": "Unset",
      "Imax": "pass",
      "Imin": "pass",
      "dI": "5.000",
      "dQ": "0.000",
      "dQM": "0.000",
      "dt (s)": "1.0000",
      "dta (s)": "0.1000",
      "goto Ns'": "0",
      "name": "Chronoamperometry / Chronocoulometry",
      "nc cycles": "0",
      "record": "I",
      "ti (h:m:s)": "1:15:0.0000",
      "unit Imax": "mA",
      "unit Imin": "mA",
      "unit dI": "mA",
      "unit dQ": "mA.h",
      "unit dQM": "mA.h",
      "vs.": "Ref",
    },
  },
}
`;
